platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/FelixConfig.java

```diff
@@ -0,0 +1,173 @@
+     */
+    String getPid() {
+        return config.getPid();
+    }
+
+    /**
+     * @return the File object pointing to the this configuration on disk.
+     */
+    @Nullable
+    File getFelixFile() {
+        return felixFile;
+    }
+
+    /**
+     * Sets the felix file name using the default technique from the config repo, using .config in
+     * case complex data structures are present.
+     *
+     * @throws IOException if an error occurs persisting to config admin.
+     */
+    void setFelixFile() throws IOException {
+        setFelixFile(createFelixFileName(config));
+    }
+
+    /**
+     * Sets the felix file name to the provided {@link File} argument.
+     *
+     * @param file the file to use for the felix file name property.
+     * @throws IOException if an error occurs persisting to config admin.
+     */
+    void setFelixFile(File file) throws IOException {
+        Dictionary<String, Object> propsWithFelixFileName = config.getProperties();
+        propsWithFelixFileName.put(FELIX_FILENAME_PROP,
+                file.toURI()
+                        .toString());
+        config.update(propsWithFelixFileName);
+
+        this.felixFile = file;
+    }
+
+    /**
+     * Determine if the felix.fileinstall.filename property changed.
+     *
+     * @param original file location to compare to the current file location poitned to by this
+     *                 config.
+     * @return False if the file locations by absolute path are identical. True otherwise.
+     */
+    boolean filePropChanged(File original) {
+        return !Objects.equals(felixFile, original);
+    }
+
+    private File extractFelixFileInstallProp(Configuration config) {
+        if (config.getProperties() == null) {
+            LOGGER.debug("Config properties are null, they may have just been created");
+            return null;
+        }
+        Object felixConfigFileName = config.getProperties()
+                .get(FELIX_FILENAME_PROP);
+        if (felixConfigFileName == null) {
+            LOGGER.debug("Felix filename prop not found for pid {}", config.getPid());
+            return null;
+        }
+        return createFileFromFelixProp(felixConfigFileName);
+    }
+
+    /**
+     * Code adopted from Karaf's Config Repository Impl:
+     * https://github.com/apache/karaf/blob/master/config/src/main/java/org/apache/karaf/config/core/impl/
+     * ConfigRepositoryImpl.java#L101-L114
+     */
+    private File createFileFromFelixProp(Object felixConfigFileName) {
+        try {
+            if (felixConfigFileName instanceof URL) {
+                return new File(((URL) felixConfigFileName).toURI());
+            }
+            if (felixConfigFileName instanceof URI) {
+                return new File((URI) felixConfigFileName);
+            }
+            if (felixConfigFileName instanceof String) {
+                return new File(new URL((String) felixConfigFileName).toURI());
+            }
```

Comment:
Correct. If Felix, Karaf (or anyone really) sets bad info as the felix file name, we treat it as if the name wasn't set at all. We just log differently. This helps keep things running smoothly despite any surprises.